cmd/controller-manager/main.go

```diff
@@ -189,14 +190,17 @@ func main() {
 	}
 	onStopped := func() {
 		klog.Error("leader election lost")
+		if err2 := srv.Shutdown(context.Background()); err2 != nil {
```

Comment:
what about just panic like before and pass contect.TODO to `leaderelection.RunOrDie`

Reply:
The reason why I removed fatal before was that it didn't work properly. 
If you are interested, you can change to fatal and try.

Reply:
I am not interested 🐶

I think it was caused by the ctx pass to `RunOrDie` is done before?
and now there are still possible to call srv.Shutdown twice, one in the signal handle func, one is here after reaching end of main func

Reply:
I think no need to collect the coverage during the leader election lost case, we just need to make sure that the coverage can be collected during the normal shut down.

Reply:
https://github.com/pingcap/tidb-operator/pull/3890/commits/e5cf3c6fe9751ab2b0bc7cbefc8188bf58ae963c @july2993 do you like it?

Reply:
> I think no need to collect the coverage during the leader election lost case, we just need to make sure that the coverage can be collected during the normal shut down.

yes, because leader election lost is not a normal case, we can treat it as failed to run the test like unit test.